pkg/controller/jobframework/setup.go

```diff
@@ -51,6 +51,9 @@ func SetupControllers(mgr ctrl.Manager, log logr.Logger, opts ...Option) error {
 		logger := log.WithValues("jobFrameworkName", name)
 		fwkNamePrefix := fmt.Sprintf("jobFrameworkName %q", name)
 
+		if options.ExternalFrameworks.Has(name) {
```

Comment:
We won't need this if we have the validation.

Reply:
This is to prevent a noop webhook being registered for the external framework (like it is for a non-enabled internal framework).  If we go to having two separate lists in the manager and not include the external framework list in `forEach`, then I agree this would not be needed.

Reply:
Let's keep them separate.